components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -189,3 +195,59 @@ func GetRandString(l int) string {
 	}
 	return string(b)
 }
+
+func RemoveStatus(sub eventingv1alpha1.Subscription) *eventingv1alpha1.Subscription {
+	desiredSub := sub.DeepCopy()
+	desiredSub.Status = eventingv1alpha1.SubscriptionStatus{}
+	return desiredSub
+}
+
+func UpdateSubscription(ctx context.Context, dClient dynamic.Interface, sub *eventingv1alpha1.Subscription) error {
+	unstructuredObj, err := toUnstructuredSub(sub)
+	if err != nil {
+		return errors.Wrap(err, "failed to convert subscription to unstructured")
+	}
+	_, err = dClient.
+		Resource(GroupVersionResource()).
+		Namespace(sub.Namespace).
+		UpdateStatus(ctx, unstructuredObj, metav1.UpdateOptions{})
+
+	return err
+}
+
+func ToSubscriptionList(unstructuredList *unstructured.UnstructuredList) (*eventingv1alpha1.SubscriptionList, error) {
+	subscriptionList := new(eventingv1alpha1.SubscriptionList)
+	subscriptionListBytes, err := unstructuredList.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(subscriptionListBytes, subscriptionList)
+	if err != nil {
+		return nil, err
+	}
+	return subscriptionList, nil
+}
+
+func toUnstructuredSub(sub *eventingv1alpha1.Subscription) (*unstructured.Unstructured, error) {
+	object, err := k8sruntime.DefaultUnstructuredConverter.ToUnstructured(&sub)
+	if err != nil {
+		return nil, err
+	}
+	return &unstructured.Unstructured{Object: object}, nil
+}
+
+func GroupVersionResource() schema.GroupVersionResource {
```

Comment:
Please rename to `SubscriptionGroupVersionResource`.

Reply:
Renamed